tests/unittests/ontology_validation_tests.py

```
@@ -0,0 +1,136 @@
+class UniqueAttributeNamesTests(unittest.TestCase):
+    def test_rejects_duplicate_names(self):
+        ao = AnnotationOntology(
+            name="test",
+            attributes=[
+                AttributeSpec(name="damage", type="str", component="dropdown"),
+                AttributeSpec(
+                    name="damage", type="bool", component="checkbox"
+                ),
+            ],
+        )
+        with self.assertRaises(ValueError) as cm:
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)